### ISSUE TITLE
Treat any `en-*` dialect as default

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#159664]
Today we seem to look for a language pack when e.g. `en-GB` is declared. But the fast path is to let it through as our default locale since we have no language pack for it as far we are concerned.

**Comment by TylerLeonhardt** [comment]
We use to have an en-GB language pack. I took it down because the localization team didn't want to maintain it. However there are folks that still want it (@Tyriar being one):

https://github.com/microsoft/vscode-loc/issues/443

I've thought about offering tooling for folks to make languages packs... en-GB would be a perfect candidate for this.

So at first glance, I don't like this idea.

**Comment by Tyriar** [comment]
What's the performance impact of this? Is it just a single request one time for en-* language packs that gets cached after that?

**Comment by bpasero** [comment]
![image](https://user-images.githubusercontent.com/900690/187706864-a051b93c-9563-426e-99c5-67243b75fa8a.png)

Around `15-20ms` on my Windows machine. We mainly seem to be using `require` calls, so the stack ends up in node.js.

**Comment by Tyriar** [comment]
@TylerLeonhardt is there a way this could be cached so it's only done on a cold start or when some change is detected?

**Comment by dbaeumer** [comment]
The require call you are seeing is here: https://github.com/microsoft/vscode/blob/70c7237d867fad780af5fa04f6136a1db73163f1/src/main.js#L563 which loads the `languagePacks.js` file which isn't huge. So I guess the time we are seeing is actually spent in `getNLSConfiguration()` which tries to find a language pack on disk in case the locale is not `en` or `en-US` which we can extend to `en-*`. So we can't avoid that code for locales !== 'en'. 

If we write the locale into `argv.json` we still need to spend that time. We could simply start doing it a little earlier. @bpasero do you now the time we spent between having the content of the argv.json file and the app ready event?

**Comment by bpasero** [comment]
`app.on("ready")` is mystical to us and can be anything from milliseconds to seconds: https://github.com/microsoft/vscode/issues/22087

We track it in Kusto and I think its typically in the `100ms` range.

**Comment by dbaeumer** [comment]
OK. Good to know.

**Comment by TylerLeonhardt** [comment]
I'm fine with treating all `en` locales like `en` but if we _do_ get to a point where we have nice tooling around crafting language packs, I will want to remove that optimization to enable `en-GB` and all the rest.

**Comment by bpasero** [comment]
Yeah that was my suggestion. 

Nevertheless, if we can revisit how NLS is blocking startup in a waterfall or not, I am also happy.

**Comment by Tyriar** [comment]
Is there any reason `vs/base/node/languagePacks` isn't included in the bundle if it's relatively small?

**Comment by dbaeumer** [comment]
The code is executed in the main thread and not the renderer. That is why it is not bundled into the renderer. To my 'old' knowledge we don't have a main bundle (@bpasero is this still correct). If we do we should definitely bundle it there.

**Comment by bpasero** [comment]
Yeah, at that point its just commonJS, not AMD.

**Comment by dbaeumer** [comment]
Could we use esbuild to bundle main.js? Then we don't need to use AMD.

**Comment by bpasero** [comment]
We can for the `languagePacks.js` but it would not help for:

https://github.com/microsoft/vscode/blob/a5c0f6c1919e429b29f584aa42f5b209344d2f08/src/vs/base/node/languagePacks.js#L73-L82

**Comment by dbaeumer** [comment]
Agree, but we could easily replace this with a async read since the call to `getLanguagePackConfigurations` is already in a async function.

**Comment by bpasero** [comment]
That would be very cool! To prevent sync fs reads.

**Comment by bpasero** [comment]
Opened https://github.com/microsoft/vscode/issues/159894 for that with a PR.
